lib/phlex/sgml.rb

```diff
@@ -353,15 +353,8 @@ def enable_cache_reloading?
 		false
 	end
 
-	def vanish(*args)
-		return unless block_given?
-
-		if args.length > 0
-			@_state.capturing_into(Phlex::Vanish) { yield(*args) }
-		else
-			@_state.capturing_into(Phlex::Vanish) { yield(self) }
-		end
-
+	def vanish(...)
+		capture(...)
 		nil
 	end
```

Comment:
I think this is a reasonable simplification here. It allocates a string but I think it’s fine.

Reply:
Yeah I think so. The typical use case for `vanish` is that you'd be calling it with a block that _shouldn't_ contain any direct element calls. So the performance optimization of `Phlex::Vanish` probably wasn't even being used much.